### PR TITLE
Fix sanity test for TF2.7

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Fix sanity test for TF2.7

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/container_tests/bin/testTFAddons [hunk 1]
```diff
@@ -13,7 +13,7 @@ elif [[ $(python -c 'import tensorflow as tf; print(tf.__version__)') == 2.2.* ]
 else
   pip install -U --no-deps tensorflow_addons
 fi
-pip install typeguard
+pip install typeguard packaging
```
Comment (tejaschumbalkar) [inline test/dlc_tests/container_tests/bin/testTFAddons:16]: `packaging` is not explicitly installed in the prod image. Please suggest if this should go in the dockerfile

File: test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py [hunk 2]
```diff
@@ -211,6 +211,7 @@ def run_data_service_test(ec2_connection, ec2_instance_ami, tensorflow_training,
     python_invoker = get_python_invoker(ec2_instance_ami)
     _, tensorflow_version = test_utils.get_framework_and_version_from_tag(tensorflow_training)
     ec2_connection.run(f"{python_invoker} -m pip install --upgrade pip")
+    ec2_connection.run(f"{python_invoker} -m pip install 'protobuf>=3.20,<3.21'")
```
Comment (tejaschumbalkar) [inline test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py:214]: This was not required while releasing TF2.7 earlier. Please suggest if this change should also b applied in the dockerfile

File: dlc_developer_config.toml [hunk 3]
```diff
@@ -20,7 +20,7 @@ benchmark_mode = false
 [build]
```
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:20]: To be reverted before PR merge

File: tensorflow/buildspec.yml [hunk 4]
```diff
@@ -1,8 +1,8 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
```
Comment (tejaschumbalkar) [inline tensorflow/buildspec.yml:1]: To be reverted before PR merge

File: test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py (hunk 2 above)
Comment (matherit) [reply test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py:214]: I believe this change should be in the dockerfile. This was not required previously as 3.20.1 was the highest released version (excluding release candidates) until May 25th. https://pypi.org/project/protobuf/#history

And our images seem incompatible at this time with the recently released version.

File: test/dlc_tests/container_tests/bin/testTFAddons (hunk 1 above)
Comment (matherit) [reply test/dlc_tests/container_tests/bin/testTFAddons:16]: Looking through the repo, it looks like we have not been installing this in our TF images, instead relying on our test helpers to install it. I think leaving it out of the docker file is fine.

 For what it's worth, we have been installing it in some of our PT and HF images, so putting it in the image wouldn't be unique.
 
 That being said, since it hasn't been in our TF images and is used only for helping with tests, I'd say we leave it out of our docker file.

File: test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py (hunk 2 above)
Comment (saimidu) [reply test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py:214]: Would it be possible to take on an exploratory task (low priority) to rework this test so that it runs TFContainer - TFContainer, rather than TFContainer - Host? It might be a long-term fix for a test that requires a server-client setup with both installing TF-dependencies.

Same for the TF MNIST inference test as well.

**Review by saimidu — APPROVED** [review]

**Review by matherit — APPROVED** [review]
